wtss_plugin/wtss_qgis.py

```diff
@@ -693,110 +687,46 @@ def display_point(self, pointTool):
             self.dlg.input_longitude.setValue(x)
             self.dlg.input_latitude.setValue(y)
         try:
-            self.getLayers()
-            self.selected_location = {
-                'long' : x,
-                'lat' : y,
-                'layer_name' : str(self.layer.name()),
-                'crs' : str(self.layer.crs().authid())
-            }
-            history_key = str(
-                (
-                    "[{long:,.7f}, {lat:,.7f}]"
-                ).format(
-                    long = self.selected_location.get('long'),
-                    lat = self.selected_location.get('lat')
-                )
-            )
-            self.locations[history_key] = self.selected_location
-            locations_keys = list(self.locations.keys())
-            self.dlg.history_list.clear()
-            self.dlg.history_list.addItems(locations_keys)
-            self.dlg.history_list.setCurrentRow(len(locations_keys) - 1)
+            self.save_on_history(x, y)
             self.draw_point(x, y)
         except AttributeError:
             pass
 
     def addCanvasControlPoint(self, enable):
         """Generate a canvas area to get mouse position."""
-        crs_id = "4326"
         self.point_tool = None
+        self.pan_map = None
```

Comment:
Do you really need to keep it as member attribute?
Could it be just a local variable instead?